unittest/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+# Add test that depends on the eigenpy library
+# import eigenpy is not authorized in those tests
+function(eigenpy_add_test name)
+  jrl_boostpy_add_module(${name} ${name}.cpp)
+  target_link_libraries(${name} PRIVATE eigenpy)
+  jrl_target_set_default_compile_options(${name} PRIVATE)
+  jrl_target_enforce_msvc_conformance(${name} PRIVATE)
+
+  jrl_target_set_output_directory(${name} OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${name})
+
+  jrl_python_generate_init_py(
+      ${name}
+      OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/${name}/__init__.py
+  )
+
+  add_test(
+    NAME "Eigenpy: import ${name}"
+    COMMAND $<TARGET_FILE:Python::Interpreter> -c "import ${name}"
+  )
+
+  set(test_name "Eigenpy: test_${name}.py")
+  add_test(
+    NAME ${test_name}
+    COMMAND $<TARGET_FILE:Python::Interpreter> ${CMAKE_CURRENT_SOURCE_DIR}/test_${name}.py
+  )
+  set_property(
+    TEST ${test_name}
+    PROPERTY
+      ENVIRONMENT_MODIFICATION
+        "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:eigenpy>/.."
+        "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:${name}>/.."
+  )
+endfunction()
+
+eigenpy_add_test(matrix)
+eigenpy_add_test(type_info)
+eigenpy_add_test(multiple_registration)
+if(BUILD_TESTING_SCIPY)
+  eigenpy_add_test(sparse_matrix)
+endif()
+eigenpy_add_test(tensor)
+eigenpy_add_test(geometry)
+eigenpy_add_test(complex)
+eigenpy_add_test(deprecation_policy)
+eigenpy_add_test(return_by_ref)
+eigenpy_add_test(include)
+eigenpy_add_test(eigen_ref)
+eigenpy_add_test(user_type)
+eigenpy_add_test(std_vector)
+eigenpy_add_test(std_array)
+eigenpy_add_test(std_pair)
+eigenpy_add_test(std_map)
+eigenpy_add_test(user_struct)
+eigenpy_add_test(std_unique_ptr)
+eigenpy_add_test(virtual_factory)
+
+set(TEST_TYPE "boost::optional")
+set(MODNAME boost_optional)
+configure_file(optional.cpp.in ${CMAKE_CURRENT_SOURCE_DIR}/boost_optional.cpp)
```

Comment:
I'm not a huge fan of generating file in the source directory. It can be commited by mistake.
Maybe allow eigenpy_add_test to have a SOURCE_FILE optional argument that can take absolute path source file as argument.